pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -189,6 +196,235 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 	return registries, nil
 }
 
+// registryIsMatch returns true if flags in the registry entry indicate that
+// its URL might not be used as-is, but will have URLs matched against it in
+// order to selectively apply configuration flags to an unlisted URL.
+func registryIsMatch(reg Registry) bool {
+	return reg.Insecure || reg.Blocked
+}
+
+// registryMatches returns true if the candidate location matches the pattern.
+// Recognized patterns include:
+// * IPv4[/class][:port][/path]
+// * IPv6[/class][:port][/path]
+// * hostname[:port][/path]
+// where IPv4, IPv6, and hostname may include wildcard characters.  The path
+// is implicitly terminated with "/", and individual components may also
+// include wildcard characters.
+func registryMatches(pattern, candidate string) (bool, error) {
+	defaultRegistryPort := "5000"
+	phost, pport, ppath, err := parseMatchPattern(pattern)
+	if err != nil {
+		return false, err
+	}
+	chost, cport, cpath, err := parseMatchCandidate(candidate)
+	if err != nil {
+		return false, err
+	}
+	// Check the host component
+	_, pipnet, perr := net.ParseCIDR(phost)
+	cip := net.ParseIP(chost)
+	if cip == nil && perr == nil {
+		// can't match a non-IP against a CIDR address
+		return false, nil
+	}
+	if perr == nil {
+		// pattern is CIDR and the candidate is an IP
+		if !pipnet.Contains(cip) {
+			return false, nil
+		}
+	} else {
+		// pattern and candidate are both names
+		hosts, err := path.Match(phost, chost)
+		if err != nil {
+			return false, fmt.Errorf("error matching candidate host %q against pattern host %q: %v", chost, phost, err)
+		}
+		if !hosts {
+			return false, nil
+		}
+	}
+	// Check the ports, if specified, or default ports
+	if pport == "" {
+		pport = defaultRegistryPort
+	}
+	if cport == "" {
+		cport = defaultRegistryPort
+	}
+	if pport != "*" && pport != cport {
+		return false, nil
+	}
+	// Compare paths, only so far as the pattern supplies one
+	pdir, ppath := path.Split(ppath)
+	cdir, cpath := path.Split(cpath)
+	for pdir != "" {
+		if cdir == "" {
+			return false, nil
+		}
+		matches, err := path.Match(pdir, cdir)
```

Comment:
Why introduce this at all? “just in case”? I can’t see how this can work with the `Prefix`/`URL` mapping.